packages/react-server-cli/src/run.js

```diff
@@ -30,7 +29,11 @@ export default function run(options = {}) {
 		// Pass. Commands need to check for routes themselves.
 	}
 
-	options.outputUrl = jsUrl || `${httpsOptions ? "https" : "http"}://${host}:${jsPort}/`;
+	// The non-jsUrl created output URL should be as relative as possible.  '//' ensures that we use the same
+	// scheme as the browser, which is important when using a fronting server that terminates HTTPS connections and
+	// proxies requests to react-server over HTTP.  Another note, is that `host` should be `0.0.0.0` to make it relative
+	// to the hostname being used.
+	options.outputUrl = jsUrl || `//${host}:${jsPort}/`;
```

Comment:
Nope, it's included in the `webpack-dev-server` and `webpack-hot-middleware` client packages though:

https://github.com/webpack/webpack-dev-server/blob/master/client/index.js#L113-L121